### ISSUE TITLE
[Bug]: Not clearly addressed to the assistant

### ISSUE THREAD

**Issue by sikc231** [zeroclaw-labs/zeroclaw#6002]
### Affected component

runtime/daemon

### Severity

S1 - workflow blocked

### Current behavior

zeroClaw in container querying llama.cpp (locally).
**How it works:**

1. Im sending "Hi" message to telegram.
2. zeroClaw catches telegram message and sends it to llama.cpp
3. llama.cpp processes it (very slow - running only on CPU Ryzen 3400G)
4. llama.cpp server finishes processing
5. zeroclaw returns the below

**Config.toml (only providers.models):**
```
[providers]
fallback = "llamacpp"

[providers.models]

[providers.models.llamacpp]
base_url = "http://192.168.0.221:8080/v1"
merge_system_into_user = false
model = "bartowski/Meta-Llama-3.1-8B-Instruct-GGUF:Q4_K_M"
requires_openai_auth = false
temperature = 0.7
timeout_secs = 1200
```

**Error from zeroClaw:**
```
zclaw-alpha  |   💬 [telegram] from 7399291831: hi
zclaw-alpha  |   ⏳ Processing message...
zclaw-alpha  | 2026-04-22T14:02:43.627677Z  INFO zeroclaw_channels::orchestrator: ⏱ Memory recall completed mem_recall_ms=0 sender_empty=true group_empty=true
zclaw-alpha  |   🤖 No reply (358208ms): Not clearly addressed to the assistant
```

llama.cpp server logs:
`main: server is listening on http://0.0.0.0:8080
main: starting the main loop...
srv  update_slots: all slots are idle
srv  log_server_r: done request: GET /v1/chat/completions 172.18.0.2 404
srv  params_from_: Chat format: peg-native
slot get_availabl: id  3 | task -1 | selected slot by LRU, t_last = -1
srv  get_availabl: updating prompt cache
srv          load:  - looking for better prompt, base f_keep = -1.000, sim = 0.000
srv        update:  - cache state: 0 prompts, 0.000 MiB (limits: 8192.000 MiB, 16384 tokens, 8589934592 est)
srv  get_availabl: prompt cache update took 0.01 ms
slot launch_slot_: id  3 | task -1 | sampler chain: logits -> ?penalties -> ?dry -> ?top-n-sigma -> top-k -> ?typical -> top-p -> min-p -> ?xtc -> temp-ext -> dist
slot launch_slot_: id  3 | task 0 | processing task, is_child = 0
slot update_slots: id  3 | task 0 | new prompt, n_ctx_slot = 16384, n_keep = 0, task.n_tokens = 2579
slot update_slots: id  3 | task 0 | n_tokens = 0, memory_seq_rm [0, end)
slot update_slots: id  3 | task 0 | prompt processing progress, n_tokens = 2048, batch.n_tokens = 2048, progress = 0.794106
srv  params_from_: Chat format: peg-native
srv          stop: cancel task, id_task = 0
srv  log_server_r: done request: POST /v1/chat/completions 172.18.0.2 200
srv          stop: cancel task, id_task = 2
srv  log_server_r: done request: POST /v1/responses 172.18.0.2 200
srv  params_from_: Chat format: peg-native
slot      release: id  3 | task 0 | stop processing: n_tokens = 2048, truncated = 0
slot get_availabl: id  3 | task -1 | selected slot by LCP similarity, sim_best = 0.794 (> 0.100 thold), f_keep = 1.000
slot launch_slot_: id  3 | task -1 | sampler chain: logits -> ?penalties -> ?dry -> ?top-n-sigma -> top-k -> ?typical -> top-p -> min-p -> ?xtc -> temp-ext -> dist
slot launch_slot_: id  3 | task 5 | processing task, is_child = 0
slot update_slots: id  3 | task 5 | new prompt, n_ctx_slot = 16384, n_keep = 0, task.n_tokens = 2579
slot update_slots: id  3 | task 5 | n_tokens = 2048, memory_seq_rm [2048, end)
slot init_sampler: id  3 | task 5 | init sampler, took 0.45 ms, tokens: text = 2579, total = 2579
slot update_slots: id  3 | task 5 | prompt processing done, n_tokens = 2579, batch.n_tokens = 531
slot print_timing: id  3 | task 5 |
prompt eval time =   89132.21 ms /   531 tokens (  167.86 ms per token,     5.96 tokens per second)
       eval time =    2322.59 ms /    10 tokens (  232.26 ms per token,     4.31 tokens per second)
      total time =   91454.79 ms /   541 tokens
slot      release: id  3 | task 5 | stop processing: n_tokens = 2588, truncated = 0
srv  update_slots: all slots are idle
srv  log_server_r: done request: POST /v1/chat/completions 172.18.0.2 200`

### Expected behavior

ZeroClaw should return me the response to my telegram message

### Steps to reproduce

```bash
1. Im sending "Hi" message to telegram.
2. zeroClaw catches telegram message and sends it to llama.cpp
3. llama.cpp processes it (very slow - running only on CPU Ryzen 3400G)
4. llama.cpp server finishes processing
5. zeroclaw returns the below
```

### Logs / stack traces

```text
main: server is listening on http://0.0.0.0:8080 main: starting the main loop... srv  update_slots: all slots are idle srv  log_server_r: done request: GET /v1/chat/completions 172.18.0.2 404 srv  params_from_: Chat format: peg-native slot get_availabl: id  3 | task -1 | selected slot by LRU, t_last = -1 srv  get_availabl: updating prompt cache srv          load:  - looking for better prompt, base f_keep = -1.000, sim = 0.000 srv        update:  - cache state: 0 prompts, 0.000 MiB (limits: 8192.000 MiB, 16384 tokens, 8589934592 est) srv  get_availabl: prompt cache update took 0.01 ms slot launch_slot_: id  3 | task -1 | sampler chain: logits -> ?penalties -> ?dry -> ?top-n-sigma -> top-k -> ?typical -> top-p -> min-p -> ?xtc -> temp-ext -> dist slot launch_slot_: id  3 | task 0 | processing task, is_child = 0 slot update_slots: id  3 | task 0 | new prompt, n_ctx_slot = 16384, n_keep = 0, task.n_tokens = 2579 slot update_slots: id  3 | task 0 | n_tokens = 0, memory_seq_rm [0, end) slot update_slots: id  3 | task 0 | prompt processing progress, n_tokens = 2048, batch.n_tokens = 2048, progress = 0.794106 srv  params_from_: Chat format: peg-native srv          stop: cancel task, id_task = 0 srv  log_server_r: done request: POST /v1/chat/completions 172.18.0.2 200 srv          stop: cancel task, id_task = 2 srv  log_server_r: done request: POST /v1/responses 172.18.0.2 200 srv  params_from_: Chat format: peg-native slot      release: id  3 | task 0 | stop processing: n_tokens = 2048, truncated = 0 slot get_availabl: id  3 | task -1 | selected slot by LCP similarity, sim_best = 0.794 (> 0.100 thold), f_keep = 1.000 slot launch_slot_: id  3 | task -1 | sampler chain: logits -> ?penalties -> ?dry -> ?top-n-sigma -> top-k -> ?typical -> top-p -> min-p -> ?xtc -> temp-ext -> dist slot launch_slot_: id  3 | task 5 | processing task, is_child = 0 slot update_slots: id  3 | task 5 | new prompt, n_ctx_slot = 16384, n_keep = 0, task.n_tokens = 2579 slot update_slots: id  3 | task 5 | n_tokens = 2048, memory_seq_rm [2048, end) slot init_sampler: id  3 | task 5 | init sampler, took 0.45 ms, tokens: text = 2579, total = 2579 slot update_slots: id  3 | task 5 | prompt processing done, n_tokens = 2579, batch.n_tokens = 531 slot print_timing: id  3 | task 5 | prompt eval time =   89132.21 ms /   531 tokens (  167.86 ms per token,     5.96 tokens per second) eval time =    2322.59 ms /    10 tokens (  232.26 ms per token,     4.31 tokens per second) total time =   91454.79 ms /   541 tokens slot      release: id  3 | task 5 | stop processing: n_tokens = 2588, truncated = 0 srv  update_slots: all slots are idle srv  log_server_r: done request: POST /v1/chat/completions 172.18.0.2 200
```

### ZeroClaw version

b8858

### Operating system

Ubuntu Server 24

### Regression?

No, first-time setup

### Pre-flight checks

- [x] I reproduced this on the latest master branch or latest release.
- [x] I redacted secrets, tokens, and personal data from all submitted content.

**Comment by TeoConnexioh** [comment]
This does not appear to be a Telegram or llama.cpp transport bug.

The message "Not clearly addressed to the assistant" is produced by the ZeroClaw reply-intent precheck, which classified the inbound DM as "NO_REPLY" before normal response generation. So the provider finished, but ZeroClaw intentionally suppressed the visible reply.

This is better classified as a reply-intent false negative or UX behavior issue than a runtime or provider failure.

**Comment by sikc231** [comment]
```
zclaw-alpha  |   🤖 Reply (6817ms): [Task failed — not continuing this request]
zclaw-alpha  |   💬 [telegram] from 7399291831: hi
zclaw-alpha  |   ⏳ Processing message...
zclaw-alpha  | 2026-04-23T12:58:55.862749Z  INFO zeroclaw_channels::orchestrator: ⏱ Memory recall completed mem_recall_ms=0 sender_empty=true group_empty=true
zclaw-alpha  | 2026-04-23T12:59:01.326455Z  INFO zeroclaw_channels::orchestrator: ⏱ Starting LLM call elapsed_before_llm_ms=5464
zclaw-alpha  | 2026-04-23T12:59:02.623868Z  INFO zeroclaw_channels::orchestrator: ⏱ LLM call completed llm_call_ms=1297 total_ms=6761
zclaw-alpha  |   🤖 Reply (6762ms): [Task failed — not continuing this request]
zclaw-alpha  |   💬 [telegram] from 7399291831: hi assistant
zclaw-alpha  |   ⏳ Processing message...
zclaw-alpha  | 2026-04-23T12:59:07.923085Z  INFO zeroclaw_channels::orchestrator: ⏱ Memory recall completed mem_recall_ms=0 sender_empty=true group_empty=true
zclaw-alpha  | 2026-04-23T12:59:13.304076Z  INFO zeroclaw_channels::orchestrator: ⏱ Starting LLM call elapsed_before_llm_ms=5382
zclaw-alpha  | 2026-04-23T12:59:14.197980Z  INFO zeroclaw_channels::orchestrator: ⏱ LLM call completed llm_call_ms=893 total_ms=6276
zclaw-alpha  |   🤖 Reply (6276ms): [No output]
```

**Comment by Audacity88** [comment]
**Potentially related:** #6034 — same symptom (provider reports "No user query found in messages"). Both involve local LLM providers (llama.cpp / Qwen) receiving conversation history with user messages missing. Could be the same history construction issue or a provider-specific message formatting bug.

**Comment by singlerider** [comment]
Punting to v0.8.1: this is a vague, hard-to-reproduce llama.cpp-over-Telegram report without a clear defect isolated to the v0.8.0 config or multi-agent paths, so it needs triage before it can gate a release.

**Comment by singlerider** [comment]
@sikc231 following up here so we can move this forward. The diagnosis from @TeoConnexioh and @Audacity88 points at the reply-intent precheck classifying your inbound DM as `NO_REPLY` before generation, not a Telegram or llama.cpp transport fault.

To reproduce and confirm that path, two things would help:

1. The exact inbound message text that got dropped (the raw string you sent, verbatim), since the precheck is sensitive to phrasing and whether the bot is addressed directly.
2. Your `reply_intent` configuration from `config.toml` (the whole `[reply_intent]` block if present, or confirmation that it is unset and running defaults).

If you can also confirm whether this only happens in 1:1 DMs versus group chats, that isolates the addressing heuristic. Marking this `needs-author-action` until we have those. See #6034 for a related "No user query found" report that may share a root cause.

**Comment by Audacity88** [comment]
## Status Update

This still needs the details requested on June 15: the exact inbound message, the `[reply_intent]` configuration or confirmation that it is unset, and whether the behavior occurs in a direct message, a group, or both.

Marking this stale under the issue lifecycle policy. If the behavior still occurs on a current release, please add those details by August 3, 2026. Otherwise this may close after the response window and can be reopened later with a current reproduction.

**Comment by Audacity88** [comment]
## Status Update

No current reproduction or requested configuration details were added by the August 3 deadline, and no implementation PR is open. Closing this under the issue stale policy.

This can be reopened with a current reproduction that includes the exact dropped message, the active `reply_intent` configuration, and whether it occurs in direct messages, groups, or both.
